### PR TITLE
Canonicalize avg_pool2d singleton lists during export

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #185591

`aten.avg_pool2d` accepts `int[2]` arguments for `kernel_size`, `stride`,
and `padding`, and eager execution permits a single-element list or tuple by
expanding it internally. `torch.export` records the proxy node before the
native/meta implementation performs that expansion, so models using
`kernel_size=(2,)` or `[2]` could produce an ExportedProgram containing
`aten.avg_pool2d.default(x, [2])`. That IR disagrees with the operator schema
and breaks downstream consumers that expect the recorded argument to already be
the canonical two-element form.

The fix canonicalizes only the proxy arguments used to create FX nodes for
`aten.avg_pool2d.default`, expanding singleton list/tuple values for
`kernel_size`, `stride`, and `padding` while preserving empty stride as the
operator default. The actual eager call still receives the original arguments,
so runtime behavior is unchanged. Earlier PR #154353 documented the eager
behavior; this patch fixes the exported IR side.

Fixes #153149
Generated by my agent

Benchmark Results
- Command: microbenchmark repeated `torch.export.export` calls for a 100-op
  relu chain and a direct `aten.avg_pool2d.default(x, [2])` module, 3 warmup
  iterations and 12 timed iterations per case.
- Before (main): relu_chain_100_ops median 0.071131s; avg_pool2d_singleton
  median 0.005553s.
- After (fix-153149): relu_chain_100_ops median 0.060361s;
  avg_pool2d_singleton median 0.005231s.
- Raw timings are recorded in state/153149/notes.

Test Plan:
- python test/export/test_export.py TestExport.test_avg_pool2d_export_single_element_args
- python test/test_nn.py TestNNDeviceTypeCPU.test_avg_pool2d_invalid_parameters_cpu
- lintrunner -a

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says this PR is being closed as part of a cleanup of open agentic PRs to focus review bandwidth on a smaller set; the linked issue remains open.
